CSVSource/src/main/java/io/vantiq/extsrc/CSVSource/CSVReader.java

```diff
@@ -96,44 +238,58 @@ static String setFieldName(int i, Map<String,String>schema)
             processNullValues = Boolean.parseBoolean(config.get("processNullValues").toString());
         }
 
+        boolean skipFirstLine = false;
+        if (config.get("skipFirstLine") != null) {
+            skipFirstLine = Boolean.parseBoolean(config.get("skipFirstLine").toString());
+        }
+
         int MaxLinesInEvent = (int) config.get(MAX_LINES_IN_EVENT);
-        ArrayList<Map<String,String>> file = new ArrayList<Map<String,String>>(); 
+        ArrayList<Map<String, String>> file = new ArrayList<Map<String, String>>();
 
         try (BufferedReader br = new BufferedReader(new FileReader(csvFile))) {
-            numOfRecords = 0 ; 
+            numOfRecords = 0;
             while ((line = br.readLine()) != null) {
-                
-                // use comma as separator
-                String[] values = line.split(delimiter);
-                Map<String, String> lineValues = new HashMap<String, String>(); 
-                
-                int schemaFieldIndex = 0 ; 
-                for (int i = 0 ; i < values.length ; i++) {
-                    if (values[i].length() != 0 ) {
-                        String currField = setFieldName(schemaFieldIndex,schema);
-                        lineValues.put(currField, values[i]);
-                        schemaFieldIndex++;
-                    } else if ( processNullValues) {
-                        schemaFieldIndex++;
+
+                if (!skipFirstLine) {
+                    // use comma as separator
+                    String[] values = line.split(delimiter);
+                    Map<String, String> lineValues = new HashMap<String, String>();
+
+                    int schemaFieldIndex = 0;
+                    for (int i = 0; i < values.length; i++) {
+                        if (values[i].length() != 0) {
+                            String currField = setFieldName(schemaFieldIndex, schema);
+                            lineValues.put(currField, values[i]);
+                            schemaFieldIndex++;
+                        } else if (processNullValues) {
+                            schemaFieldIndex++;
+                        }
                     }
-                }
 
-                file.add(lineValues);
-                numOfRecords++; 
-                
-                if (file.size() >= MaxLinesInEvent ) {
-                    sendNotification(csvFile, packetIndex, file, oClient);
-                    file = new ArrayList<Map<String,String>>();
-                    packetIndex++;
+                    file.add(lineValues);
+                    numOfRecords++;
+
+                    if (file.size() >= MaxLinesInEvent) {
+                        log.info("TX Packet {} Size {} Total num of Records {}", packetIndex, MaxLinesInEvent,
```

Comment:
a lot of `.info()` logging.  I'd suggest `.debug()` and one info/file if required.

Reply:
it doesn;t dump the content , just the a single statement . i still using it .